client/store/theme.js

```diff
@@ -164,20 +170,44 @@ export const state = () => ({
 });
 
 export const getters = {
+  theme: state => state.theme,
   themes: state => state.themes,
-  themeVariables: (state, value) => {
-    state.themes.find(el => value === el.value);
-  }
+  themeVariables: state => state.themes.find(el => state.theme === el.value)
 };
 
 export const mutations = {
+  setTheme(state, theme) {
+    if (state.themes.find(e => e.value === theme)) {
+      state.theme = theme;
+    }
+  },
+  // TODO: implement Error Handling
   cloneTheme(state, parentThemeName, newThemeName) {
     let parentTheme;
     if (!state.themes.find(e => newThemeName === e.value)) {
-      if ((parentTheme = state.themes.find(e => parentThemeName === e.value))) {
+      if ((parentTheme = state.themes.find(e => parentThemeName === e.value)) != undefined) {
         parentTheme.value = newThemeName;
+        parentTheme.default = false;
         state.themes.push(parentTheme);
-        // TODO: verify if this if is legit
+      }
+    }
+  },
+  editTheme(state, theme) {
+    let themeInArray;
+    if ((themeInArray = state.themes.find(e => theme.value === e.value)) != undefined) {
```

Comment:
Can you also check for the type? Like this:
```suggestion
    if ((themeInArray = state.themes.find(e => theme.value === e.value)) !== undefined) {
```